### PR TITLE
Search instrument from api

### PR DESCRIPTION
**Pull request by Yuan-EPAM**
- using `iexcloud` api for finance data
- searchResults coming from redux

### CODE REVIEW

File: frontend-opal/src/components/SearchResult/SearchResult.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import styles from './styles/index';
+
+import ResultLists from '../../container/SearchResult/ResultLists';
+import SearchHeader from '../../container/SearchResult/SearchHeader';
+
+function Index() {
+  return (
+    <View style={styles.Index}>
+      <SearchHeader />
+      <ResultLists />
```
Comment (adamgyulavari) [inline frontend-opal/src/components/SearchResult/SearchResult.js]: @Yuan-EPAM technically this is going to be only one `ResultList` or multiple `Results`

File: frontend-opal/src/thunks/searchResult.js [hunk 2]
```diff
@@ -1,33 +1,113 @@
...
+//     name: 'MSFT',
+//     fullName: 'Microsoft Corp.',
+//     market: 'NASDAQ'
+//   },
+//   {
+//     name: 'FBMS',
+//     fullName: 'First Branchshares Inc/MS',
+//     market: 'NASDAQ'
+//   }
+// ];
```
Comment (adamgyulavari) [inline frontend-opal/src/thunks/searchResult.js]: @Yuan-EPAM whopsie

File: frontend-opal/src/thunks/searchResult.js [hunk 3]
```diff
@@ -1,33 +1,113 @@
...
+
+async function fetchData() {
+  const symbols = `https://cloud.iexapis.com/stable/ref-data/symbols?token=${IEXCLOUD_PUBLIC_KEY}`;
```
Comment (adamgyulavari) [inline frontend-opal/src/thunks/searchResult.js]: @Yuan-EPAM extract this into a `stockApi` file and separate the `baseUrl` as a constant

File: frontend-opal/src/thunks/searchResult.js [hunk 4]
```diff
@@ -1,33 +1,113 @@
...
+async function extractTopNResult(dataSet, num) {
+  const res = [];
+  // searchRes.first: 100% matched result; secondRes.second: partially matched results
+  res.push(...dataSet.first);
+  if (dataSet.second.length > 0) {
+    const count = getExtractNum(dataSet, num);
+    res.push(...dataSet.second.slice(0, count - 1));
   }
-];
+  return res;
```
Comment (adamgyulavari) [inline frontend-opal/src/thunks/searchResult.js]:
```javascript
async function extractTopNResult({first, second}, num) {
  return [...first, ...second].slice(0, num);
}
```

File: frontend-opal/src/thunks/searchResult.js [hunk 5]
```diff
@@ -1,33 +1,113 @@
...
+    results: []
+  };
+  return async dispatch => {
+    if (input !== '') {
+      const searchRes = await searchData(input);
+      const res = mapTitle(searchRes);
+
+      dispatch(
+        actions.search({
+          input,
```
Comment (adamgyulavari) [inline frontend-opal/src/thunks/searchResult.js]: @Yuan-EPAM you don't need to send the input, it's not needed in the reducer at least

File: frontend-opal/src/thunks/searchResult.js [hunk 6]
```diff
@@ -1,33 +1,113 @@
...
+    res.push(tmp);
+  });
+  return res;
+}
+
+function search(input) {
+  const initRes = {
+    input,
+    results: []
+  };
```
Comment (adamgyulavari) [inline frontend-opal/src/thunks/searchResult.js]: @Yuan-EPAM if this is not a const, you can remove the 2 dispatches and just have one after the if

File: frontend-opal/src/thunks/searchResult.js (hunk 4 above)
Comment (Yuan-EPAM) [reply frontend-opal/src/thunks/searchResult.js]: Wow, only 1 line!!!

I feel shame of my codes :rofl:

Comment (adamgyulavari) [reply frontend-opal/src/thunks/searchResult.js]: yeah well, a little overcomplicated it :)

File: frontend-opal/src/thunks/searchResult.js (hunk 6 above)
Comment (Yuan-EPAM) [reply frontend-opal/src/thunks/searchResult.js]: refactor done!

File: frontend-opal/src/thunks/searchResult.js [hunk 7]
```diff
@@ -90,24 +53,19 @@ function mapTitle(data) {
...
-          results: res
-        })
-      );
-    } else {
-      dispatch(actions.search(initRes));
+      initRes = mapTitle(searchRes);
     }
+    dispatch(
+      actions.search({
+        results: initRes
```
Comment (adamgyulavari) [inline frontend-opal/src/thunks/searchResult.js]: @Yuan-EPAM are you sure to dispatch this action payload:
```javascript
{
  results: {
     results: [...]
  }
}
```

Comment (adamgyulavari) [reply frontend-opal/src/thunks/searchResult.js]: also don't call it initRes, it is the result :)

Comment (Yuan-EPAM) [reply frontend-opal/src/thunks/searchResult.js]: Ops, strange thing. Actually in my local file, the dispatch function was `dispatch(actions.search(initRes)`. 

And how about the name of  `data`~

File: frontend-opal/src/thunks/searchResult.js [hunk 8]
```diff
@@ -1,33 +1,66 @@
...
+  return async dispatch => {
+    const data = {
+      results: []
+    };
+    if (input !== '') {
+      const searchRes = await searchData(input);
+      data.results = mapTitle(searchRes);
+    }
+
+    dispatch(actions.search(data));
```
Comment (adamgyulavari) [inline frontend-opal/src/thunks/searchResult.js]: @Yuan-EPAM how about
```javascript
let results = [];
...
dispatch(actions.search({results}));

Comment (Yuan-EPAM) [reply frontend-opal/src/thunks/searchResult.js]: Got it!

**Pull request merged**